sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/HttpResponse.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2024 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.javasdk;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class HttpResponse {
+
+  private final StatusCode.Success statusCode;
+  private final String contentType;
+  private final byte[] body;
+
+  private HttpResponse(StatusCode.Success statusCode, String contentType, byte[] body) {
+    this.statusCode = statusCode;
+    this.contentType = contentType;
+    this.body = body;
+  }
+
+  private HttpResponse(StatusCode.Success statusCode) {
+    this(statusCode, "application/octet-stream", new byte[0]);
+  }
+
+  public StatusCode.Success getStatusCode() {
+    return statusCode;
+  }
+
+  public String getContentType() {
+    return contentType;
+  }
+
+  public byte[] getBody() {
+    return body;
+  }
+
+  public static HttpResponse ok() {
+    return new HttpResponse(StatusCode.Success.OK);
+  }
+
+  public static HttpResponse created() {
+    return new HttpResponse(StatusCode.Success.CREATED);
+  }
+
+  public static HttpResponse noContent(StatusCode.Success statusCode) {
+    return new HttpResponse(statusCode);
+  }
+
+  public static HttpResponse ok(String text) {
+    if (text == null) throw new IllegalArgumentException("text must not be null");
+    return new HttpResponse(StatusCode.Success.OK, "text/plain", text.getBytes(UTF_8));
+  }
+
+  public static HttpResponse ok(Object object) {
+    if (object == null) throw new IllegalArgumentException("object must not be null");
```

Comment:
I would answer the same as above.

Reply:
I agree for byte[], but not for Object. Json spec accept defines `null` as a valid value (or should I say non-value?). 

Although we might find it a bad idea, it might be something that we should allow.